src/js/tech/middleware.js

```diff
@@ -0,0 +1,63 @@
+import Tech from './tech.js';
+import toTitleCase from '../utils/to-title-case.js';
+
+const middlewares = {};
+
+export function use(type, middleware) {
+  (middlewares[type] = middlewares[type] || []).push(middleware);
```

Comment:
If we're picking this nit, I generally prefer `concat` to `|| []`.

```
middlewares[type] = [].concat(middlewares[type], middlware)
```

This is a little bit of nonsense here since we're ok with mutating the original array, so ¯\\\_(ツ)_/¯